apps/src/blockly/addons/cdoUtils.ts

```diff
@@ -764,3 +764,39 @@ export function toolboxWithoutIds(
   removeIdsFromBlocks(toolboxDom);
   return Blockly.Xml.domToText(toolboxDom);
 }
+
+// Sets the lab code based on the student's blocks and any extra (e.g. initialization) code.
+// The students blocks are considered to be any on the main or hidden workspaces.
+export function getAllGeneratedCode(extraCode?: string) {
+  let code = extraCode || '';
```

Comment:
do we need a `\n` after `extraCode`?

Reply:
It doesn't seem like it. In the example level with the moon and stars above, this is the code we end up with:

```
"\n\n\n\n\n\n\nTurtle.penColour('#ffffff', 'block_id_draw-color');\n\n\n\nfunction draw_a_star() {\n    executionInfo.checkTimeout(); if (executionInfo.isTerminated()){return;}\n  for (var count = 0; count < 8; count++) {\n      executionInfo.checkTimeout(); if (executionInfo.isTerminated()){return;}\n    Turtle.moveForward(25, 'block_id_,tQB)s0?x}Be~N-pJzb2');\n    Turtle.moveBackward(25, 'block_id_f1I=YG6VeM5.6XcQP?~t');\n    Turtle.turnRight(45, 'block_id_cNG}r$iGr2!s2%ebX1nA');\n  }\n}\n\n\n\ndraw_a_star();\n\n\n"
```

which amounts to this:
```







Turtle.penColour('#ffffff', 'block_id_draw-color');



function draw_a_star() {
    executionInfo.checkTimeout(); if (executionInfo.isTerminated()){return;}
  for (var count = 0; count < 8; count++) {
      executionInfo.checkTimeout(); if (executionInfo.isTerminated()){return;}
    Turtle.moveForward(25, 'block_id_,tQB)s0?x}Be~N-pJzb2');
    Turtle.moveBackward(25, 'block_id_f1I=YG6VeM5.6XcQP?~t');
    Turtle.turnRight(45, 'block_id_cNG}r$iGr2!s2%ebX1nA');
  }
}



draw_a_star();


```

I think we're already super generous with new lines.